processes/src/main/java/pl/mrasoft/springboottest/jbmp/internal/SpringQuartzSchedulerService.java

```diff
@@ -0,0 +1,441 @@
+            ProcessJobContext processCtx = (ProcessJobContext) ctx;
+            jobname = processCtx.getSessionId() + "-" + processCtx.getProcessInstanceId() + "-" + processCtx.getTimer().getId();
+            if (processCtx instanceof StartProcessJobContext) {
+                jobname = "StartProcess-"+((StartProcessJobContext) processCtx).getProcessId()+ "-" + processCtx.getTimer().getId();
+            }
+        } else if (ctx instanceof NamedJobContext) {
+            jobname = ((NamedJobContext) ctx).getJobName();
+        } else {
+            jobname = "Timer-"+ctx.getClass().getSimpleName()+ "-" + id;
+
+        }
+        logger.debug("Scheduling timer with name " + jobname);
+        // check if this scheduler already has such job registered if so there is no need to schedule it again        
+        try {
+            JobDetail jobDetail = scheduler.getJobDetail(jobKey(jobname, "jbpm"));
+
+            if (jobDetail != null) {
+                TimerJobInstance timerJobInstance = (TimerJobInstance) jobDetail.getJobDataMap().get("timerJobInstance");
+                return timerJobInstance.getJobHandle();
+            }
+        } catch (SchedulerException e) {
+
+        }
+        GlobalQuartzJobHandle quartzJobHandle = new GlobalQuartzJobHandle(id, jobname, "jbpm");
+        TimerJobInstance jobInstance = globalTimerService.
+                getTimerJobFactoryManager().createTimerJobInstance( job,
+                ctx,
+                trigger,
+                quartzJobHandle,
+                (InternalSchedulerService) globalTimerService );
+        quartzJobHandle.setTimerJobInstance( (TimerJobInstance) jobInstance );
+
+        interceptor.internalSchedule(jobInstance);
+        return quartzJobHandle;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public boolean removeJob(JobHandle jobHandle) {
+        GlobalQuartzJobHandle quartzJobHandle = (GlobalQuartzJobHandle) jobHandle;
+
+        try {
+
+            boolean removed =  scheduler.deleteJob(jobKey(quartzJobHandle.getJobName(), quartzJobHandle.getJobGroup()));
+            return removed;
+        } catch (SchedulerException e) {
+
+            throw new RuntimeException("Exception while removing job", e);
+        } catch (RuntimeException e) {
+            SchedulerMetaData metadata;
+            try {
+                metadata = scheduler.getMetaData();
+                if (metadata.getJobStoreClass().isAssignableFrom(JobStoreCMT.class)) {
+                    return true;
+                }
+            } catch (SchedulerException e1) {
+
+            }
+            throw e;
+        }
+    }
+
+    @Override
+    public void internalSchedule(TimerJobInstance timerJobInstance) {
+
+        GlobalQuartzJobHandle quartzJobHandle = (GlobalQuartzJobHandle) timerJobInstance.getJobHandle();
+        // Define job instance
+        JobDetailImpl jobq = new JobDetailImpl(quartzJobHandle.getJobName(), quartzJobHandle.getJobGroup(), QuartzJob.class);
```

Comment:
To jest deprecated, ale nie chciałem robić niewiadomo ile zmian więc nie używałem buildera.